src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java

```diff
@@ -40,15 +40,52 @@ public enum ReplicateStatusCause {
     DATASET_FILE_BAD_CHECKSUM,
 
     // computation
-    PRE_COMPUTE_FAILED,
-    POST_COMPUTE_FAILED,
-    TEE_NOT_SUPPORTED,
     APP_NOT_FOUND_LOCALLY,
+    APP_COMPUTE_FAILED,
+    // pre-computation
+    PRE_COMPUTE_TASK_ID_MISSING, //exit 3
+    PRE_COMPUTE_EXIT_REPORTING_FAILED, //exit 2
+    PRE_COMPUTE_OUTPUT_PATH_MISSING, //exit 1 (known)
+    PRE_COMPUTE_IS_DATASET_REQUIRED_MISSING,
+    PRE_COMPUTE_DATASET_URL_MISSING,
+    PRE_COMPUTE_DATASET_KEY_MISSING,
+    PRE_COMPUTE_DATASET_CHECKSUM_MISSING,
+    PRE_COMPUTE_DATASET_FILENAME_MISSING,
+    PRE_COMPUTE_INPUT_FILES_NUMBER_MISSING,
+    PRE_COMPUTE_AT_LEAST_ONE_INPUT_FILE_URL_MISSING,
+    PRE_COMPUTE_OUTPUT_FOLDER_NOT_FOUND,
+    PRE_COMPUTE_DATASET_DOWNLOAD_FAILED,
+    PRE_COMPUTE_INVALID_DATASET_CHECKSUM,
+    PRE_COMPUTE_DATASET_DECRYPTION_FAILED,
+    PRE_COMPUTE_SAVING_PLAIN_DATASET_FAILED,
+    PRE_COMPUTE_INPUT_FILE_DOWNLOAD_FAILED,
+    PRE_COMPUTE_FAILED_UNKNOWN_ISSUE, //exit 1 (unknown)
+    // Post compute
+    POST_COMPUTE_UNKNOWN_ISSUE,
+    POST_COMPUTE_MISSING_TASK_ID,
+    POST_COMPUTE_MISSING_WORKER_ADDRESS,
+    POST_COMPUTE_MISSING_TEE_CHALLENGE_PRIVATE_KEY,
+    POST_COMPUTE_MISSING_ENCRYPTION_PUBLIC_KEY,
+    POST_COMPUTE_MISSING_STORAGE_TOKEN,
+    POST_COMPUTE_COMPUTED_FILE_NOT_FOUND,
+    POST_COMPUTE_RESULT_DIGEST_COMPUTATION_FAILED,
+    POST_COMPUTE_OUT_FOLDER_ZIP_FAILED,
+    POST_COMPUTE_ENCRYPTION_FAILED,
+    POST_COMPUTE_RESULT_FILE_NOT_FOUND,
+    POST_COMPUTE_DROPBOX_UPLOAD_FAILED,
+    POST_COMPUTE_IPFS_UPLOAD_RAILED,
+    POST_COMPUTE_INVALID_TEE_SIGNATURE,
+    POST_COMPUTE_SEND_COMPUTED_FILE_FAILED,
```

Comment:
Several feedbacks on the Enum.
It lacks a bit of consistency between POST and PRE compute.
Some examples:
`PRE_COMPUTE_FAILED_UNKNOWN_ISSUE`, `POST_COMPUTE_UNKNOWN_ISSUE`
`POST_COMPUTE_MISSING_TASK_ID`, `PRE_COMPUTE_TASK_ID_MISSING`
`POST_COMPUTE_OUT_FOLDER`, `PRE_COMPUTE_OUTPUT_FOLDER`
...

I'm ok with either option as long as we have a better consistency and a kind of naming convention.

Small typo in `POST_COMPUTE_IPFS_UPLOAD_RAILED`

Reply:
Yes, I agree and plan to update, thanks for reminding it :)

Reply:
Thanks for pointing them out, @mcornaton updated them. Closing this comment.